Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -553,20 +554,21 @@ void MatchTOF::propagateTPCTracks(int sec)
     }
 
     if (trc.getX() < o2::constants::geom::XTPCOuterRef - 1.) {
-      if (!propagateToRefX(trc, o2::constants::geom::XTPCOuterRef, 10, intLT0) || TMath::Abs(trc.getZ()) > Geo::MAXHZTOF) { // we check that the propagat>
+      if (!propagateToRefX(trc, o2::constants::geom::XTPCOuterRef, 10, intLT0dummy) || TMath::Abs(trc.getZ()) > Geo::MAXHZTOF) { // we check that the propagat>
         mNotPropagatedToTOF[trkType::UNCONS]++;
         continue;
       }
     }
 
-    o2::base::Propagator::Instance()->estimateLTFast(intLT0, trc);
-
     // the "rough" propagation worked; now we can propagate considering also the cov matrix
-    if (!propagateToRefX(trc, mXRef, 2, intLT0)) { // || TMath::Abs(trc.getZ()) > Geo::MAXHZTOF) { // we check that the propagation with the cov matrix w>
+    if (!propagateToRefX(trc, mXRef, 2, intLT0dummy)) { // || TMath::Abs(trc.getZ()) > Geo::MAXHZTOF) { // we check that the propagation with the cov matrix w>
       mNotPropagatedToTOF[trkType::UNCONS]++;
       continue;
     }
 
+    // calculate track length from now on starting with fast estimate
+    o2::base::Propagator::Instance()->estimateLTFast(intLT0, trc);
+
```

Comment:
then this will not be needed anymore.

Reply:
Ok, I will change it and re-run again a test. Thanks

Reply:
done